New Transaction class. New bpi code (incomplete)

ruicovelo opened the pull request:
I've added a new Transaction class and started adding BPINet code. More will follow soon. The new class adds a level of abstraction I think was missing but this breaks apps/shell.py. Code for other banks must be changed first. 

Even though the code is not complete, I'm requesting a pull so someone else can try and integrate the new Transaction class in other banks code. I cannot do this easily because I don't have an account in any other bank.

Please bear with me cause I'm new to github, python and only a "part-time" coder.

ruicovelo committed:
Added bpi.py which contains/will contain the code for accessing Banco
BPI.
Changed apps/shell.py and __init__.py to use the new bpi.py.

As far as bpi.py is concerned, it's an early stage:
- done login / authentication code
- done get_account_list (needs a tweak)
- started get_movements but is not parsing yet

ruicovelo committed:
Added bpi.py which contains/will contain the code for accessing Banco
BPI.
Changed apps/shell.py and __init__.py to use the new bpi.py.

As far as bpi.py is concerned, it's an early stage:
- done login / authentication code
- done get_account_list (needs a tweak)
- started get_movements but is not parsing yet

ruicovelo committed:
Add code to test BPI access

ruicovelo committed:
Add code to test BPI access

ruicovelo committed:
Now gets most recent transactions for BPI

ruicovelo committed:
Now gets most recent transactions for BPI

ruicovelo committed:
added new Transaction class. Changed BPI code to use new class. Other banks code must be changed to use new class.

ruicovelo committed:
added new Transaction class. Changed BPI code to use new class. Other banks code must be changed to use new class.

ruicovelo committed:
BPI code changed to use new Transaction class.
This breaks app/shell.py.
Other banks code must be changed to use new transation class.

ruicovelo committed:
BPI code changed to use new Transaction class.
This breaks app/shell.py.
Other banks code must be changed to use new transation class.

jneves commented:
Cool, this goes ahead with a part of my todo-list of defining transactions :).

I'll pull it tomorrow, probably to a new branch, while I adapt the code for the other banks.

Thanks a lot!

Commit:
LOGINSTARTPAGE no longer works so changed to just the login page

ruicovelo committed:
- Corrected bug parsing values for BPI (added parse_value() to
transaction)
- get_movements() updated to get movements according to specified
dates

ruicovelo committed:
- Corrected bug parsing values for BPI (added parse_value() to
transaction)
- get_movements() updated to get movements according to specified
dates

Commit:
Account number parsing fix

Commit:
Account number parsing fix

ruicovelo committed:
Adding a basic example about how you would use mymoney to simply get the
most recent transactions.

ruicovelo committed:
Adding a basic example about how you would use mymoney to simply get the
most recent transactions.

ruicovelo committed:
Preparing for adding proxy support. Not tested.

ruicovelo committed:
Preparing for adding proxy support. Not tested.

ruicovelo committed:
Parsing fix

ruicovelo committed:
Parsing fix

ruicovelo committed:
FIX: local vs static variables and __str__ function

ruicovelo committed:
FIX: local vs static variables and __str__ function

ruicovelo committed:
Adding new FileExport class for exporting transaction into files

ruicovelo committed:
Adding new FileExport class for exporting transaction into files

ruicovelo committed:
small comment fix

ruicovelo committed:
small comment fix

ruicovelo committed:
Adding first version QIFFileExport compatible with gnucash

ruicovelo committed:
Adding first version QIFFileExport compatible with gnucash

ruicovelo committed:
Prepared for custom date format

ruicovelo committed:
Prepared for custom date format

ruicovelo committed:
Merge branch 'master' of https://github.com/ruicovelo/mymoney
Rebasing with Niko Roberts patch for cgd.py

ruicovelo committed:
First version for python 3.4.
Tested under Windows. Requires more testing.

ruicovelo committed:
FIX for 3.4

ruicovelo committed:
Comment fix

ruicovelo committed:
First version support for ticket restaurant

ruicovelo committed:
Merge branch 'python3'

ruicovelo committed:
Merge branch 'ticketrestaurant'

ruicovelo committed:
FIX: negative values for withdrawls, positive values for deposits

ruicovelo committed:
Adding support for euroticket REST API. This version is based on https://github.com/marmelo/euroticket-alacard-rest-api.
I have found this hardly ever works but it does not seem related to this code but something server side. I am not sure.

ruicovelo committed:
Adding requirements.txt file

ruicovelo committed:
Fix syntax

ruicovelo committed:
New login

ruicovelo committed:
get_movements now working for most recent transactions

ruicovelo committed:
Some code cleanup and linting fixes. No longer support for getting account list.

ruicovelo committed:
More linting fixes

ruicovelo committed:
Removed unused library

ruicovelo committed:
Removing debug code

ruicovelo committed:
Merge branch 'new-bpinet'

ruicovelo committed:
Removed not working get_account_list implementation

ruicovelo committed:
Removed support for multiple account

ruicovelo committed:
Santander first working version that gets most recent transactions

ruicovelo committed:
Code cleanup

ruicovelo committed:
Support for iterating through recent movements pages.
Get card list and card statement list.

ruicovelo committed:
Fixed URLs
Changed default iter_pages to False to avoid excessive calls by default
Moved get_card_statetment_list to Card class
Added iter_pages to Card.get_movements
Fixed parse_value to support " EUR" in value

ruicovelo committed:
Update README.md

ruicovelo committed:
Merge pull request #4 from ruicovelo/santander

Santander

ruicovelo committed:
Merge branch 'master' of github.com:ruicovelo/mymoney

ruicovelo committed:
First functional version for MyEdenred

ruicovelo committed:
Update README.md

ruicovelo committed:
Merge pull request #5 from ruicovelo/myedenred

First functional version for MyEdenred

ruicovelo committed:
Some linting fixes

ruicovelo committed:
Fixing last commit ffs

ruicovelo committed:
Merge branch 'myedenred' of github.com:ruicovelo/mymoney into myedenred

ruicovelo committed:
Merge branch 'master' of github.com:ruicovelo/mymoney

ruicovelo committed:
Merge branch 'myedenred'

ruicovelo committed:
Adding "requests" as requirement

ruicovelo committed:
my date import

ruicovelo committed:
Add filter by since date support and small bug fix

ruicovelo committed:
Add since_date filter to card get movements and fix page iteration

ruicovelo committed:
Fix >= since_date

ruicovelo committed:
Add support for ssl verification bypass for debugging purposes

ruicovelo committed:
Fixing login form name change. Adding missing headers to better simulate a browser.

ruicovelo committed:
Merge branch 'master' of github.com:ruicovelo/mymoney

ruicovelo committed:
Add *.json

ruicovelo committed:
Simpler .gitignore

ruicovelo committed:
json

ruicovelo committed:
get_cc_movements for getting credit card movements

ruicovelo committed:
Revert "json"

This reverts commit 59e6311578a7e5cbd6c9c58550ce3f299aaac44e.

ruicovelo committed:
Ignore DS_Store and vscode

ruicovelo committed:
No longer saves cookie file unless explicitly requested with persist_cookies

ruicovelo committed:
Move to package directory structure

ruicovelo committed:
pyproject.toml initial version

ruicovelo committed:
get_balance version 1.0

ruicovelo committed:
Minor changes

ruicovelo committed:
Linting fixes

ruicovelo committed:
Update version
